src/ocm/components/clusterWizard/wizardTransition.ts

```diff
@@ -37,18 +46,48 @@ export const getClusterWizardFirstStep = (
     }
     return 'static-ip-network-wide-configurations';
   }
+  const step = getStepIdByHostValidations(hosts);
   switch (state) {
     case 'ready':
       return 'review';
     case 'pending-for-input':
     case 'adding-hosts':
     case 'insufficient':
-      return 'host-discovery';
+      return step ? step : 'host-discovery';
     default:
       return 'cluster-details';
   }
 };
 
+const getStepIdByHostValidations = (hosts: Host[] | undefined) => {
+  const failingValidations: HostValidationId[] = getHostValidations(hosts);
+  let step = '';
+  failingValidations.forEach((failingValidationId) => {
+    const stepFound = findValidationStep<string>(
+      { validationId: failingValidationId },
```

Comment:
why do you need to create an object?
It's enough `findValidationStep<string>(failingValidationId, wizardStepsValidationsMap)`